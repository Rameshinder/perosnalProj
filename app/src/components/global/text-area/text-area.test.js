import React from 'react'
import {shallow, mount} from 'enzyme'

import TextArea from './index'

describe('Heading', () => {
  const app = shallow(<TextArea
    rows={2}
    cols={2}
    maxlength={250}
    placeholder={'Place text'}
    onFocus={jest.fn()}
    onBlur={jest.fn()}
    onChange={jest.fn()} textAreaFunc={jest.fn()} autoFocus />)

  const wrapper = shallow(<TextArea
    rows={2}
    cols={2}
    placeholder={'Place text'}
    onFocus={jest.fn()}
    onBlur={jest.fn()}
    onChange={jest.fn()} textAreaFunc={jest.fn()} autoFocus />)

  test('renders properly', () => {
    expect(app).toMatchSnapshot()
  })
  test('renders properly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('TextArea should render without crashing', () => {
    shallow(<TextArea
      rows={2}
      cols={2}
      maxlength={250}
      placeholder={'Place text'}
      onFocus={jest.fn()}
      onBlur={jest.fn()}
      onChange={jest.fn()}
      autoFocus />)
  })

  test('on focus event for textarea component', () => {
    const handleFocus = jest.fn()
    handleFocus()
    app.instance().handleFocus()
    expect(handleFocus).toHaveBeenCalled()
  })

  test('on blur event for textarea component', () => {
    const handleBlur = jest.fn()
    handleBlur()
    app.instance().handleBlur()
    expect(handleBlur).toHaveBeenCalled()
  })

  test('onChange function to be called for textarea component', () => {
    const handleChange = jest.fn()
    handleChange('hi')
    app.instance().handleChange('hi')
    expect(handleChange).toHaveBeenCalled()
  })
  test('TextArea componentDidMount', () => {
    let wrapper = mount(<TextArea
      rows={2}
      cols={2}
      maxlength={250}
      placeholder={'Place text'}
      onFocus={jest.fn()}
      onBlur={jest.fn()}
      onChange={jest.fn()}
      hasTransition
      transitionTiming={500}
      media={{ currentTime: 0, duration: 500 }}
      timeout={(fn, _) => fn()}
      ref='textArea' />)
    jest.useFakeTimers()
    const componentDidMount = jest.fn()
    componentDidMount()
    wrapper.instance().componentDidMount()
    jest.runAllTimers()
    expect(componentDidMount).toHaveBeenCalled()
  })
})
