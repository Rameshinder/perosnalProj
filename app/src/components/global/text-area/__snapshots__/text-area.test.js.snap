// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextArea
    autoFocus={true}
    cols={2}
    maxlength={250}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Place text"
    rows={2}
    textAreaFunc={[Function]}
    transitionTiming={500}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.textarea
          autoFocus={undefined}
          cols={2}
          data-test-id="undefined-textarea"
          id="undefined-textarea"
          maxLength={250}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Place text"
          rows={2}
          title={undefined}
        />,
        <styled.label
          for="undefined-textarea"
        >
          250
        </styled.label>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": undefined,
          "cols": 2,
          "data-test-id": "undefined-textarea",
          "id": "undefined-textarea",
          "maxLength": 250,
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "placeholder": "Place text",
          "rows": 2,
          "title": undefined,
        },
        "ref": "textArea",
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": 250,
          "for": "undefined-textarea",
        },
        "ref": null,
        "rendered": 250,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.textarea
            autoFocus={undefined}
            cols={2}
            data-test-id="undefined-textarea"
            id="undefined-textarea"
            maxLength={250}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Place text"
            rows={2}
            title={undefined}
          />,
          <styled.label
            for="undefined-textarea"
          >
            250
          </styled.label>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": undefined,
            "cols": 2,
            "data-test-id": "undefined-textarea",
            "id": "undefined-textarea",
            "maxLength": 250,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "placeholder": "Place text",
            "rows": 2,
            "title": undefined,
          },
          "ref": "textArea",
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": 250,
            "for": "undefined-textarea",
          },
          "ref": null,
          "rendered": 250,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Heading renders properly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextArea
    autoFocus={true}
    cols={2}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Place text"
    rows={2}
    textAreaFunc={[Function]}
    transitionTiming={500}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.textarea
          autoFocus={undefined}
          cols={2}
          data-test-id="undefined-textarea"
          id="undefined-textarea"
          maxLength={250}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Place text"
          rows={2}
          title={undefined}
        />,
        <styled.label
          for="undefined-textarea"
        >
          250
        </styled.label>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": undefined,
          "cols": 2,
          "data-test-id": "undefined-textarea",
          "id": "undefined-textarea",
          "maxLength": 250,
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "placeholder": "Place text",
          "rows": 2,
          "title": undefined,
        },
        "ref": "textArea",
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": 250,
          "for": "undefined-textarea",
        },
        "ref": null,
        "rendered": 250,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.textarea
            autoFocus={undefined}
            cols={2}
            data-test-id="undefined-textarea"
            id="undefined-textarea"
            maxLength={250}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Place text"
            rows={2}
            title={undefined}
          />,
          <styled.label
            for="undefined-textarea"
          >
            250
          </styled.label>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": undefined,
            "cols": 2,
            "data-test-id": "undefined-textarea",
            "id": "undefined-textarea",
            "maxLength": 250,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "placeholder": "Place text",
            "rows": 2,
            "title": undefined,
          },
          "ref": "textArea",
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": 250,
            "for": "undefined-textarea",
          },
          "ref": null,
          "rendered": 250,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
